File: mobile/src/main/java/net/mm2d/android/util/ActivityUtils.java
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.android.util;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.ActivityOptionsCompat;
import android.view.View;

/**
 * @author <a href="mailto:ryo@mm2d.net">大前良介 (OHMAE Ryosuke)</a>
 */
public class ActivityUtils {
    public static Bundle makeScaleUpAnimationBundle(@NonNull final View v) {
        return ActivityOptionsCompat.makeScaleUpAnimation(v, 0, 0, v.getWidth(), v.getHeight())
                .toBundle();
    }
}

File: mobile/src/main/java/net/mm2d/android/util/ActivityUtils.kt
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.android.util

import android.os.Bundle
import android.view.View
import androidx.core.app.ActivityOptionsCompat

/**
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
object ActivityUtils {
    fun makeScaleUpAnimationBundle(v: View): Bundle? = ActivityOptionsCompat
        .makeScaleUpAnimation(v, 0, 0, v.width, v.height)
        .toBundle()
}
